modules/programs/hyprpaper.nix

```diff
@@ -0,0 +1,108 @@
+{ pkgs, config, lib, ... }:
+
+with lib;
+
+let cfg = config.programs.hyprpaper;
+in {
+  meta.maintainers = [ lib.maintainers.silmarp ];
+
+  options = {
+    programs.hyprpaper = {
+      enable = mkEnableOption "hyprpaper";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.hyprpaper;
+        defaultText = literalExpression "pkgs.hyprpaper";
+        description = "The package to use for the hyprpaper binary.";
+      };
+
+      systemd = {
+        enable =
+          mkEnableOption "Whether to enable hyprpaper systemd integration.";
+
+        target = mkOption {
+          type = types.str;
+          default = "graphical-session.target";
+          example = "hyprland-session.target";
+          description = ''
+            The systemd target that will automatically start the Hyprpaper service.
+          '';
+        };
+      };
+
+      settings = with types; {
```

Comment:
This combination of `settings` and `extraConfig` seems a bit unconventional.

Perhaps make settings a [freeform module](https://nixos.org/manual/nixos/stable/#sec-freeform-modules) with `preload` and `wallpapers` being explicitly defined. Then there would be no need for an `extraConfig` option.